### DISCUSSION TITLE
One connection pool per host, not per database

### DISCUSSION THREAD

**Discussion by anton-johansson** [prisma/prisma#21505] [Q&A]
### Question

I'm running a multi-tenant application. I have two Microsoft SQL servers, with a bunch of databases on 'em. I currently create a new connection pool for each individual database. But it feels like a lot of resources for this. I would love to have one connection pool per SQL server instead. I'm guessing it should be easy to just switch database before each query somehow.

Is there any way of handling this with Prisma in a good way? Maybe omitting the database name from the connection URL, and run some `USE [MyDatabase];` before each query?

**Comment by ludralph** [comment]
Hi @anton-johansson 👋 

Thank you for raising this question.

> Is there any way of handling this with Prisma in a good way? Maybe omitting the database name from the connection URL, and run some USE [MyDatabase]; before each query?

I would recommend you to programmatically change the connection URL at runtime. This would still create multiple connection pools, but this way, you would avoid hardcoding the database name in the schema.

https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#programmatically-override-a-datasource-url

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by anton-johansson** [reply]
Hey @ludralph, thanks for the reply!

I already create Prisma clients and set the connection URL programmatically. There are many databases, so it wouldn't be doable otherwise.

This does absolutely not answer my question, since it does not really consider my question at all. :smile: The question is about reducing the number of connections created, saving resources. I only need a pool of X connections per database server. It actually takes quite some time to create these connections, so I would like to reduce them, both for startup time and for resource consumption.

What I need is someway of switching database before I perform a query.

```ts
const tenant = 1;
// tenant 1 has database name Database1
// tenant 2 has database name Database2
// ... etc

const client = await getDatabaseClient(tenant);

// Here, I need some hook to automatically run `USE Database1` on the actual connection that I was handed
const data = await client.myTable.findMany();
```

**Comment by janpio** [comment]
That is currently not possible. We treat the database (in MySQL), or schema (in PostgreSQL and others) as the boundary that you connect to.

What you are looking for would probably be implemented as part of https://github.com/prisma/prisma/issues/2077 - so maybe 👍 that and subscribe to updates there.

**Reply by anton-johansson** [reply]
Hey @janpio!

Thanks for that reference, I've given my thumbs up!

I see that it has been open for a few years. How should I manage my expectations? Is this something that is being discussed for the roadmap in the near future? Otherwise, we might need to look elsewhere. We're currently in progress of onboarding more tenants, and the number of tables in our model will only grow as well.

I also found this, which describes the issue very well too:
https://github.com/prisma/prisma/issues/20375

**Reply by janpio** [reply]
Sorry, can't tell you when we will get to that.

**Reply by anton-johansson** [reply]
Allright, thanks for the update! Please keep us in the loop here. :pray:
